OpenUpSA/wazimap-ng-ui
# Dev Tools for easier testing

## Description

A dev tools for easier testing of different BE. 
First use-case is for changing the hostname
install it only if not production coming from the awesome [K.C. Dodds](https://kentcdodds.com/blog/make-your-own-dev-tools)
it should be kept out of the bundle when the bundle is built with `NODE_ENV=production` otherwise it will install it but only show on dev or when `dev-tools=true` is set as a query param

## Related Issue
__no issue created_

## How to test it locally
run the app, open the browser. you should see in the bottom left a small settings icon. hover over it, it should open the dev tools pane. select a different hostname. 
if it's not showing add `dev-tools=true` in the query string. 

run the app with `NODE_ENV=production` and it should not show the settings icon.


https://deploy-preview-100--wazimap-staging.netlify.app/?dev-tools=true

## Screenshots
<img width="336" alt="Bildschirmfoto 2020-10-05 um 07 45 12" src="https://user-images.githubusercontent.com/688980/95044618-28b00f80-06e0-11eb-8542-78e675bfd9ac.png">
<img width="1169" alt="Bildschirmfoto 2020-10-05 um 07 45 05" src="https://user-images.githubusercontent.com/688980/95044620-29e13c80-06e0-11eb-84ef-8352edf7f339.png">


## Changelog

### Added
* added `svelte-dev-tools` component 
### Updated
* changed hostname to use `process.env.HOSTNAME` or query string

### Removed


## Checklist


### Pull Request


### Commits


### Code Quality


### Testing
